### PR TITLE
fix for using public key, if public key used disable tlog

### LINKED ISSUE #11: bug: no certificate found on signature error for using public key to verify signature
**Description**

I notice that when I try to verify the signature by using a public key, the verify sub-command still tries to verify the signature with a transparency log. 

Steps to reproduce:
```shell
$ cosign generate-key-pair
$ kubectl run alpine --image=alpine \
    --restart='Never' \
    --dry-run=client -oyaml -- sh -c "sleep 600" > alpine.yaml
$ kubectl sigstore sign -f alpine.yaml \
      -k cosign.key \
      -i gcr.io/$(gcloud config get-value project)/hello-k8s-manifest-sigstore:v1 # succeed and alpine.yaml.signed file generated
$ kubectl sigstore verify -f alpine.yaml.signed -k cosign.pub
error occured during signature verification: error occured while verifying image `gcr.io/developerguy-311909/hello-k8s-manifest-sigstore:v1`; no matching signatures:
no certificate found on signature
```

> cosign version: 0.8.3

### PR DESCRIPTION
**Pull request by developer-guy**
Signed-off-by: Batuhan Apaydın <batuhan.apaydin@trendyol.com>

This PR aims to fix #11.

### CODE REVIEW

File: pkg/k8smanifest/verify.go [hunk 1]
```diff
@@ -47,18 +47,25 @@ type SignatureVerifier interface {
 }
 
 func NewSignatureVerifier(objYAMLBytes []byte, imageRef string, pubkeyPath *string) SignatureVerifier {
+	i := &ImageSignatureVerifier{onMemoryCacheEnabled: true}
 	var annotations map[string]string
 	if imageRef == "" {
 		annotations = k8smnfutil.GetAnnotationsInYAML(objYAMLBytes)
 		if annoImageRef, ok := annotations[ImageRefAnnotationKey]; ok {
 			imageRef = annoImageRef
+			i.imageRef = imageRef
```
Comment (hirokuni-kitahara) [inline pkg/k8smanifest/verify.go:56]: could you check the location of line 56 to right after line 58 like this, please?
imageRef could come from method argument.
```
	if imageRef == "" {
		annotations = k8smnfutil.GetAnnotationsInYAML(objYAMLBytes)
		if annoImageRef, ok := annotations[ImageRefAnnotationKey]; ok {
			imageRef = annoImageRef
		}
	}
	i.imageRef = imageRef
```

**Review by hirokuni-kitahara — COMMENTED** [review]
@developer-guy Thank you very much!!
This PR is really helpful for this project.
Just one comment, could you check my comment below?
Thank you.

File: pkg/k8smanifest/verify.go (hunk 1 above)
Comment (developer-guy) [reply pkg/k8smanifest/verify.go:56]: nice one, thank you so much @hirokuni-kitahara, I updated the code according to your review.

**Review by hirokuni-kitahara — APPROVED** [review]

File: pkg/k8smanifest/verify.go (hunk 1 above)
Comment (hirokuni-kitahara) [reply pkg/k8smanifest/verify.go]: thank you very much! I have approved the changes.

Comment (yuji-watanabe-jp) [reply pkg/k8smanifest/verify.go]: thank you so much! @developer-guy @hirokuni-kitahara

Comment (developer-guy) [reply pkg/k8smanifest/verify.go]: thank you so much, folks @yuji-watanabe-jp @hirokuni-kitahara 🥳🎉

**Pull request merged**